lean4-infoview/src/infoview/info.tsx

```diff
@@ -200,6 +208,18 @@ export function InfoDisplay(props0: InfoDisplayProps) {
                     </div>
                 </Details>
             </div>
+            <div style={{display: hasWidget ? 'block' : 'none'}}>
+                <Details initiallyOpen>
+                    <summary className="mv2 pointer">
+                    Widget: {widget && widget.id}
+                      {widgetStatus === 'pending' && ' (pending)'}
+                      {widgetStatus === 'rejected' && ' (errored)'}
+                    </summary>
+                    <div className="ml1">
+                      {widget && <UserWidget pos={pos} widget={widget}/>}
```

Comment:
Ordering-wise, I think it would make more sense to put the widgets above the messages.

I'm not sure how I feel about associating widgets to source-code positions.  Does this show all widgets on the line, only the one at the exact position, or do they have a range server-side?  Most of the examples I have seen for widgets associate one widget per goal.  (But then there's also `#eval`, which kind of associates a widget to a diagnostic.)

Reply:
Sorry I should have pushed the RFC that I am writing before this.

Reply:
The idea is that widgets exist in the infotree same as any other info, and given a pos in the infoview we just render all of the widgets there, one after the other. But for most cases we would expect there to just be at most one widget.